scripts/apps/search/components/MarkedDesksInfo.jsx

```diff
@@ -35,6 +38,24 @@ export class MarkedDesksInfo extends React.Component {
 
         closeActionsMenu();
         this.renderDropdown();
+        this.setActionMenuState(true);
+    }
+
+    componentWillUnmount() {
+        if (this.state.isActionMenuOpen) {
+            closeActionsMenu();
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.state.isActionMenuOpen && nextProps.item !== this.props.item) {
+            this.setActionMenuState(false);
```

Comment:
Is this hack here because it's not appropriate to set state in this method? If so, I think we should find a better way. This is very hacky as is. Plus, we don't need this flag to be in the state. Every time you set the state this component will be re-rendered.

Instead, can you not just use a property on the class? Something like `this.isActionMenuOpen` ?

Reply:
Not a hack. You can set state in `componentwillreceiveprops` method (see https://reactjs.org/docs/react-component.html#componentwillreceiveprops).

Reply:
Still, it doesn't have to be on the state. But you seem to have accepted Petr's and my suggestion already so this will have to be removed anyway.